### ISSUE TITLE
Field Level Validation Falsely updates sync errors to Empty Object on Change Event alternatively

### ISSUE THREAD

**Issue by khpatel4991** [erikras/redux-form#2566]
I have a weird problem with field level validation. I have attached the bug behavior in the screen share. I have a Field with simple field level validation which onChange dispatches "UPDATE_SYNC_ERRORS" alternatively with correct error object and empty error object(even when the error object should not be empty). If you check the video, it'll be more clear.
The field I was tinkering with in the video is the following
```
<Field 
  name={`${packagePriceFieldName}.unit_price`}
  placeholder="Price Per Package"
  label="Unit Price"
  component={index === 0 ? StaticInput : CurrencyInput}
  seed={baseUnitPrice}
  validate={[required, currency]}
/>
```
Index is the element index in the array(in my case it would be 1)
packagePriceFieldName is the parent field name(in my case it's 'package[0].package_price[1]')
baseUnitPrice is a prop which is only used when index is 0. (so unnecessary)

When I'm tinkering with the input, even the console logs say that validation is false and still after onChange an action is dispatched with empty error object incorrectly.

The two validation functions are as follows:
```
export const required = value => value ? undefined: 'Required';
export const currency = value => {
  const validation = /^[1-9]\d*(((,\d{3}){1})?(\.\d{0,2})?)$/.test(value);
  console.log("Currency", value);
  console.log("validation", validation);
  return validation ? 
    undefined : 'Invalid Amount';
}

``` 

I also have exported the action dispatched throughout the page into json from react redux devtools.

Here's a link for the attachments.
https://drive.google.com/open?id=0ByZzhBPM_O4LNzE3cXA2QjcxaFE

**Comment by mqklin** [comment]
Why do you pass array as `validate` prop? `Field` component can take `validate` only as function: http://redux-form.com/6.5.0/docs/api/Field.md/#-validate-value-allvalues-props-error-optional-

**Comment by gustavohenke** [comment]
@mqklin, even though the `validate` prop is documented like that, the examples demonstrate that you can use it as an array: https://redux-form.com/6.5.0/examples/fieldLevelValidation

**Comment by Lavoaster** [comment]
Just a note for anyone who stumbles across this.

For some reason I had stupidly copied code across to a FieldArray component that made it a redux form, so I had a redux form in a redux form, sharing the same name. Which meant the parent form that had validation errors was then being overridden by the child form which overwrote the errors because it had none.

TL;DR Check your `reduxForm` to ensure you aren't nesting them, and that their `names` are unique.

**Comment by gustavohenke** [comment]
Because this issue saw almost no activity for a few months, I'm closing it.

**Comment by buzjuka** [comment]
When you pass field level validation as array your Field will be re-registr each time. Array wil be created each time when a component receiving new props.  You should define creating an array of validations like separate constant:

```
const neededValidations = [required, currency]

class YourComponent extends PureComponent {
  render() {
    ...
    <Field ... validate={neededValidations} />
  }
}
```

**Comment by greg606** [comment]
@boy-techy could you please provide example for this?

**Comment by danielrob** [comment]
@greg606 see #4017
